clients/drcachesim/simulator/tlb.h

```diff
@@ -46,21 +49,67 @@ namespace drmemtrace {
 
 class tlb_t : public caching_device_t {
 public:
+    tlb_t(const std::string &name = "tlb");
+
     void
     request(const memref_t &memref) override;
 
-    // TODO i#4816: The addition of the pid as a lookup parameter beyond just the tag
-    // needs to be imposed on the parent methods invalidate(), contains_tag(), and
-    // propagate_eviction() by overriding them.
+    // TODO i#4816: The addition of the pid as a lookup parameter beyond just the
+    // tag needs to be imposed on the parent methods invalidate(), contains_tag(),
+    // and propagate_eviction() by overriding them.
+
+    // Returns the replacement policy used by this TLB.
+    virtual std::string
+    get_replace_policy() const override = 0;
 
 protected:
     void
     init_blocks() override;
-
     // Optimization: remember last pid in addition to last tag
     memref_pid_t last_pid_;
 };
 
+/**
```

Comment:
(Not sure this non-external-interface header is passed to doxygen (do you see this in your local build's html output?); no harm though.)